### DISCUSSION TITLE
I have problem in my ban and kick command if I write reason: None

### DISCUSSION THREAD

**Discussion by Tasfinthebigboy** [Rapptz/discord.py#9740] [Q&A] [answered]
This is the Error message:

`Traceback (most recent call last):
  File "E:\Code\Creative-Cloud-Gen\app.py", line 63, in <module>
    @bot.tree.command(name="kick", description="Kicks member from a guild")
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Code\Creative-Cloud-Gen\.venv\Lib\site-packages\discord\app_commands\tree.py", line 887, in decorator
    command = Command(
              ^^^^^^^^
  File "E:\Code\Creative-Cloud-Gen\.venv\Lib\site-packages\discord\app_commands\commands.py", line 666, in __init__
    self._params: Dict[str, CommandParameter] = _extract_parameters_from_callback(callback, callback.__globals__)
                                                ^^^^^^^^^^^^^`

```
@bot.tree.command(name="ban", description="Ban a member from a guild perma")
async def ban_command(interaction: discord.Interaction, member: discord.Member, *, reason: None):
    try:
        await member.ban(reason=reason)
        embed = discord.Embed(title="", description="", color=discord.Color.red())
        embed.add_field(name="Banned:", value=f"{member.name} reason: {reason}", inline=False)
        await interaction.response.send_message(embed=embed)

    except:
        await interaction.response.send_message("You do not have permission to do that.", ephemeral=True)

**Comment by LostLuma** [comment, marked as answer]
You annotated the type of the `reason` argument as `None` instead of giving it a default value of `None` which it seems you intended.

Type annotations are made up of multiple parts (some of which you don't always use): `parameter: Type = Default`, so in your case you'd want to use `reason: str | None = None`, making reason an optional `str` with a default value of `None` if it's not given.

---

If you end up posting other questions please don't cut up the error in the middle (and use a large code block so the newlines work), the whole thing is useful to others when they want to help you.

**Reply by Tasfinthebigboy** [reply]
Thanks It works now
